hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified path for the log file")
+      final String logFilePathPattern) throws IOException {
+
+
+    List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+        HoodieCLI.tableMetadata.getHadoopConf()).globStatus(new Path(logFilePathPattern)))
+        .map(status -> status.getPath().toString()).collect(Collectors.toList());
+
+    Map<String, List<Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>>>> logBlockTypeToMetadata =
+        Maps.newHashMap();
+
+    for(String logFilePath: logFilePaths) {
+      FileStatus[] fsStatus = FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()).listStatus(
+          new Path(logFilePath));
+      Schema writerSchema = new AvroSchemaConverter().convert(readSchemaFromLogFile(new Path(logFilePath)));
+      HoodieLogFormat.Reader reader = HoodieLogFormat.newReader(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()),
+          new HoodieLogFile(fsStatus[0].getPath()), writerSchema, true);
+
+      //read the avro blocks
+      while (reader.hasNext()) {
+        HoodieLogBlock n = reader.next();
+        String commitTime;
+        if (n instanceof HoodieCorruptBlock) {
+          try {
+            commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+          } catch (Exception e) {
+            commitTime = "corrupt_block_1";
+            // unable to read metadta
+          }
+        } else {
+          commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+        }
+        if (logBlockTypeToMetadata.containsKey(commitTime)) {
+          logBlockTypeToMetadata.get(commitTime).add(Pair.of(n.getBlockType(), n.getLogMetadata()));
+        } else {
+          List<Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>>> list
+              = new ArrayList<>();
+          list.add(Pair.of(n.getBlockType(), n.getLogMetadata()));
+          logBlockTypeToMetadata.put(commitTime, list);
+        }
+      }
+    }
+
+    String[][] rows = new String[logBlockTypeToMetadata.size() + 1][];
+    int i = 0;
+    for(Map.Entry<String, List<Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>>>>
+        entry : logBlockTypeToMetadata.entrySet()) {
+      for(Pair<HoodieLogBlock.HoodieLogBlockType, Map<HoodieLogBlock.LogMetadataType, String>> p : entry.getValue()) {
+        String [] metadata = new String[3];
+        metadata[0] = entry.getKey();
+        metadata[1] = p.getKey().toString();
+        metadata[2] = new ObjectMapper().writeValueAsString(p.getValue());
+        rows[i] = metadata;
+        i++;
+      }
+    }
+    return HoodiePrintHelper.print(
+        new String[]{"CommitTime", "BlockType", "Metadata"}, rows);
```

Comment:
can we actually merge this method and above one into a single command that gets you recordCount (in case of data blocks) and also metadata?  Seems like it will be handy given we do the work anyway